ext/phar/tests/gh18953.phpt

```diff
@@ -0,0 +1,45 @@
+--TEST--
+Phar: Stream double free
+--EXTENSIONS--
+phar
+--INI--
+phar.readonly=0
+--ENV--
+USE_ZEND_ALLOC=0
+--FILE--
+<?php
+
+declare(strict_types=1);
+
+require __DIR__ . '/gh18953/autoload.inc';
```

Comment:
You don't need all this autoloading stuff, nor the classes at the bottom. Executing the test without these things, with ASAN+USE_ZEND_ALLOC=0, reliably reproduces the bug.